vis.go

```diff
@@ -4,10 +4,13 @@ package mtree
 import "C"
 import "fmt"
 
+// Vis is a wrapper of the C implementation of the function vis, which encodes
+// a character with a particular format/style
 func Vis(str string) (string, error) {
-	dst := new(C.char)
+	// dst needs to be 4 times the length of str (plus 1 for null char)
+	dst := C.CString(string(make([]byte, 4*len(str)+1)))
 	ret := C.strvis(dst, C.CString(str), C.VIS_WHITE|C.VIS_OCTAL|C.VIS_GLOB)
-	if ret == 0 {
```

Comment:
Does that function signature work with the pure golang logic you had
started?

On Wed, Jul 20, 2016, 23:17 Stephen Chung <notifications@github.com> wrote:

> In vis.go
> <https://github.com/vbatts/go-mtree/pull/32#discussion_r71641564>:
>
> >  	ret := C.strvis(dst, C.CString(str), C.VIS_WHITE|C.VIS_OCTAL|C.VIS_GLOB)
> > -	if ret == 0 {
>
> I believe we can just return the string
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/vbatts/go-mtree/pull/32/files/2062c5a542c24d1f758db41e5db9ff7907398324#r71641564>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEF6auvKyQ-jeN1ZMzSGFhDSzD2ZAfgks5qXuSrgaJpZM4JRXDl>
> .
>